app/models/user_account.rb

```diff
@@ -1,11 +1,16 @@
 # frozen_string_literal: true
 
 class UserAccount < ApplicationRecord
-  has_many :user_verification, dependent: :destroy
+  has_many :user_verifications, dependent: :destroy
   has_one :user_acceptable_verified_credential, dependent: :destroy
 
   validates :icn, uniqueness: true, allow_nil: true
 
+  scope :idme_verification, -> { joins(:user_verifications).merge(UserVerification.idme) }
```

Comment:
I'm not a fan of these scopes, but as I mentioned directly to you they may be a necessary evil